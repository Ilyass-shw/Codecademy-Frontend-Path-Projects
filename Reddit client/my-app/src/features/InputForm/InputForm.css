.input {
	align-self: center;
}
.input-form {
	display: flex;
	align-items: center;
	justify-content: center;
}
input {
	background-color: #121212;
	color: #d1d1d1;
	border: 1px solid #303030;
	height: 1.75rem;
	font-size: 0.9rem;
	font-weight: 545;
	width: 34.5rem;
	border-radius: 2px 0 0 2px;
	padding: 2px 6px;
}
.search-button {
	background-color: #313131;
	border-radius: 0 2px 2px 0;
	height: 2.1rem;
	width: 4.5rem;
	border-inline: none;

	border-block: 1px solid #303030;
}
input:focus {
	border: 0.2px solid rgb(0, 89, 255);
	box-shadow: 0 0 10px #313131;
}
.search-label {
	padding: 0.4rem;
	color: #ffffff;
	background-color: #5a5a5ac9;
	border-radius: 2px;
	position: relative;
	right: 65px;
	top: 45px;
	opacity: 0;
	transition: opacity 0.28s;
}
button:hover + p {
	opacity: 1;
}
