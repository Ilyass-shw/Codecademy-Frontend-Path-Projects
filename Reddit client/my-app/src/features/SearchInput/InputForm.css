.input {
	align-self: center;
}
input {
	background-color: #121212;
    color: #d1d1d1;
    border: 1px solid #303030;
}
