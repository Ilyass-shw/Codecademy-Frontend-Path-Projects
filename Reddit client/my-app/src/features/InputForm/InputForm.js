import React, { useState } from "react";
import "./InputForm.css";
// npm install --save-dev @iconify/react @iconify-icons/fluent
import { InlineIcon } from "@iconify/react";
import search16Filled from "@iconify-icons/fluent/search-16-filled";
import { fetchPosts, searchTermSet } from "../posts/postsSlice";
import { useDispatch } from "react-redux";

export const InputForm = () => {
	const dispatch = useDispatch();

	const [term, setTerm] = useState("");
	const [mouseOnSearchButton, setMouseOnSearchButton] = useState(false);

	const handleChange = ({ target }) => {
		setTerm(target.value);
	};

	const handleOnClick = () => {
		dispatch(searchTermSet(term));
		dispatch(fetchPosts());
	};

	return (
		<div className="input">
			<form action="#" className="input-form" autoComplete="off">
				<input type="text" name="search" value={term} onChange={handleChange} placeholder="Search" />
				<button
					className="search-button"
					type="button"
					onMouseEnter={() => {
						setMouseOnSearchButton(true);
					}}
					onMouseLeave={() => {
						setMouseOnSearchButton(false);
					}}
					onClick={handleOnClick}
				>
					<InlineIcon
						icon={search16Filled}
						width="1.22rem"
						height="1.22rem"
						color={mouseOnSearchButton ? "#a3a3a3" : "#6f6f6f"}
					/>
				</button>
			</form>
		</div>
	);
};
