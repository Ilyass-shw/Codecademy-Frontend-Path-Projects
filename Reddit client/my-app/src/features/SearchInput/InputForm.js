import React, { useState } from "react";
import "./InputForm.css";
// npm install --save-dev @iconify/react @iconify-icons/fluent
import { Icon, InlineIcon } from '@iconify/react';
import search16Filled from '@iconify-icons/fluent/search-16-filled';


export const InputForm = () => {
	const [term, setTerm] = useState("");

	const handleChange = ({ target }) => {
		setTerm(target.value);
	};
	return (
		<div className="input">
			<form action='#' className='input-form'>
				<input type="text" name="search" value={term} onChange={handleChange} placeholder='Search' />
				<button className='search-button'><InlineIcon icon={search16Filled} width="1.22rem" height="1.22rem" color={'#6f6f6f'}/></button>
			</form>
		</div>
	);
};
