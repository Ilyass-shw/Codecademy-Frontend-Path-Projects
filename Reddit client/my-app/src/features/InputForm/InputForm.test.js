import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import React from "react";
import { InputForm } from "./InputForm.js";
import store from "../../app/store";
import { Provider } from "react-redux";
import postsReducer from "../posts/postsSlice";
import { combineReducers, createStore } from "redux";

// function createTestStore() {
// 	const store = createStore(
// 		combineReducers({
// 			posts: postsReducer,
// 		})
// 	);
// 	return store;
// }
// const store = createTestStore();

jest.mock("@iconify-icons/fluent/search-16-filled", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});

jest.mock("react-spring/renderprops", () => {
	return { Transition: () => <div className="mocking-Transition"></div> };
});
jest.mock("@iconify/react", () => {
	return { InlineIcon: () => <p>InlineIcon mock</p> };
});

describe("InputForm", () => {
	it("should match the snapshot", () => {
		const { container } = render(
			<Provider store={store}>
				<InputForm />
			</Provider>
		);
		expect(container).toMatchSnapshot();
	});

	it("should render without crashing", () => {
		render(
			<Provider store={store}>
				<InputForm />
			</Provider>
		);
	});
});
