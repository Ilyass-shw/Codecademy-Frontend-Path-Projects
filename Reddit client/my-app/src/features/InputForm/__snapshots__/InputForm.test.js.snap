// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm should match the snapshot 1`] = `
<div>
  <div
    class="input"
  >
    <form
      action="#"
      autocomplete="off"
      class="input-form"
    >
      <div
        class="mocking-Transition"
      />
      <div
        class="button-container"
      >
        <button
          class="search-button"
        >
          <p>
            InlineIcon mock
          </p>
        </button>
        <p
          class="search-label"
        >
          Search
        </p>
      </div>
    </form>
  </div>
</div>
`;
